src/database.ts

```diff
@@ -3083,6 +3093,11 @@ class Database extends common.GrpcServiceObject {
               // Create a new data stream and add it to the end user stream.
               dataStream = this.runStream(query, options);
               dataStream.pipe(proxyStream);
+              dataStream.on('end', () => span.end());
```

Comment:
Here as well. This is only for `isSessionNotFoundError` . Lets do a simple `span.end()` here